chall01/ksappi.py

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/python3
+import sys
+
+charset = [".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..",
+	".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.",
+	"...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."]
+
+def main():
+	output = ""
+	if len(sys.argv) != 2 or len(sys.argv[1]) == 0:
+		print("usage: ./ksappi.py <a-zA-Z string>")
+		return
+	for char in sys.argv[1]:
+		if char.isalpha():
+			output += charset[ord(char.lower()) - ord('a')]
+		elif char == " ":
+			output += " "
+		else:
+			print("usage: ./ksappi.py <a-zA-Z string>")
+			return
```

Comment:
With Python there is a so-called "pythonic" way to approach tasks. I won't explain it in details here, I'll just show how this loop can be done in "pythonic" way: 
`output = "".join(charset[ord(c) - ord('a')] if c != " " else " " for c in sys.argv[1].lower())`
and if you want to make sure that everything in the string is alpha or spaces, you can do it like that:
```
if not sys.argv[1].replace(" ", "").isalpha():
    print(usage)
    return
```
you can actually just add it to the "if" above and therefore all the logic of your program would be a oneliner
`print("".join(charset[ord(c) - ord('a')] if c != " " else " " for c in sys.argv[1].lower()))`

all of this is not to say that you've done something wrong but to show perks of Python and simplicity/laconicism it provides
the main reason why Python is used so widely in research, ML, mathematics, DA, etc. is not that it's a great language even tho it is but that it allows to code and test different hypothesis super fast. and if you see that your hypothesis works, you start coding it in C++ to make it fast.